public-ui/src/app/data-browser/views/ehr-view/ehr-view-react.component.tsx

```diff
@@ -270,381 +266,402 @@ interface State {
-   }
-
-   componentDidUpdate(prevProps: Readonly<Props>) {
-        if (prevProps.domainId !== this.props.domainId) {
+export const EhrViewReactComponent = withRouteData(
+    class extends React.Component<{}, State> {
+        constructor(props: {}) {
+            super(props);
+            this.changeResults = this.changeResults.bind(this);
+            // TODO add url params and change them based on search value
+            const { search } = urlParamsStore.getValue();
+            this.state = {
+                domainId: '',
+                domain: null,
+                totalParticipants: 0,
+                title: '',
+                subTitle: '',
+                searchWord: search,
+                showStatement: false,
+                top10Results: null,
+                selectedConcept: null,
+                standardConceptIds: '',
+                numPages: null,
+                totalResults: null,
+                concepts: [],
+                standardConcepts: [],
+                currentPage: 1,
+                loading: true,
+                medlinePlusLink: null,
+                medlineTerm: null,
+                selectedMeasurementTypeFilter: false,
+                measurementTestFilter: true,
+                measurementOrderFilter: true,
+                showTopConcepts: true,
+                matchType: MatchType.NAME
+            };
+        }
+        domainTotals = _.debounce(() => {
+            this.getDomainTotals();
+            this.getTopConcepts();
+            this.changeUrl();
+        }, 1000);
+
+        componentDidMount() {
             this.getDomainTotals();
         }
-   }
 
-   changeUrl() {
-    let url = 'ehr/' + this.props.domainId;
-    if (this.state.searchWord) {
-        url += '/' + this.state.searchWord;
-    }
-    window.history.replaceState(null, 'Ehr View', url);
-   }
+        changeUrl() {
+            let url = 'ehr/' + this.state.domainId;
+            if (this.state.searchWord) {
+                url += '/' + this.state.searchWord;
+            }
+            window.history.replaceState(null, 'Ehr View', url);
+        }
 
-   getDomainTotals() {
-    const {searchWord, measurementTestFilter, measurementOrderFilter} = this.state;
-    dataBrowserApi().getDomainTotals(searchWord, measurementTestFilter ? 1 : 0, measurementOrderFilter ? 1 : 0)
+        getDomainTotals() {
+            const { searchWord, measurementTestFilter, measurementOrderFilter } = this.state;
+            dataBrowserApi().getDomainTotals(searchWord, measurementTestFilter ? 1 : 0, measurementOrderFilter ? 1 : 0)
                 .then(results => {
                     results.domainInfos.forEach(domain => {
                         const thisDomain = Domain[domain.domain];
-                        if (thisDomain && thisDomain.toLowerCase() === routeToDomain[this.props.domainId]) {
+                        if (thisDomain && thisDomain.toLowerCase() === routeToDomain[location.pathname.split('/')[2]]) {
```

Comment:
Same here, we can use `urlParamsStore.getValue().id` instead of breaking up `location.pathname`.